src/validation/obfl.rng

```diff
@@ -792,24 +792,38 @@
     <element name="table-of-contents">
       <ref name="attlist-table-of-contents"/>
       <oneOrMore>
-        <ref name="toc-entry"/>
+        <ref name="toc-block"/>
       </oneOrMore>
     </element>
   </define>
   <define name="attlist-table-of-contents" combine="interleave">
     <ref name="name-id"/>
   </define>
-  <define name="toc-entry">
-    <element name="toc-entry">
-      <ref name="attlist-toc-entry"/>
+  <define name="toc-block">
+    <element name="toc-block">
+      <ref name="attlist-toc-block"/>
       <zeroOrMore>
         <choice>
-          <ref name="blockContents"/>
           <ref name="toc-entry"/>
+          <ref name="toc-block"/>
         </choice>
       </zeroOrMore>
     </element>
   </define>
+  <define name="attlist-toc-block" combine="interleave">
+    <ref name="border-atts"/>
+    <ref name="margin-atts"/>
+    <ref name="padding-atts"/>
+    <ref name="block-atts"/>
```

Comment:
Seems a bit excesive to have ```block-atts``` for the ```toc-block```. We haven't discussed if we need an ID for the blocks but if so the ```block-id``` is reasonable. 

```list-atts``` - This block will probably only contain one ```toc-entry``` perhaps a resumed one so making a list of them seems strange.

```row-spacing``` - Seems reasonable.

```block-atts-except-list-type-row-spacing-id``` - This seems more like a normal flowing-text block attribute list. Indenting first line, underlining and alignment seems like attributes a ```toc-entry``` could have perhaps but not ```toc-block```

Reply:
@kalaspuffar Keep in mind that the current `toc-entry` (which we renamed to `toc-block`) has all of these attributes today. I agree that some are not necessary, so we could use this opportunity to drop them. But there are definitely more useful attributes than you think.

- `id`: indeed not useful
- `list-type`, `list-style`, `list-item-label`: I have to say that I don't know these that well because DAISY Pipeline doesn't make use of these, but AFAIU you can use these on any list structure, i.e. a block with a number of child blocks, and a `toc-block` can have child `toc-block`s so it applies.
- `first-line-indent`, `align`, `underline-style` etc.: these definitely apply to `toc-block`. Remember that a `toc-entry` is now just like a `span`, it has no block properties.

Another thing to keep in mind is that changing the schema ideally also requires updating the documentation so it might be easier to allow more than is strictly needed to make the documentation uncluttered.

Reply:
@bertfrees Ok, A misunderstanding on my end. I though ```toc-block``` just was a wrapper of toc-entry to facilitate other entries, not a block. But if it has the same constructs so the text could be present freely inside of a ```toc-block```, then all these make sense.

And perhaps we could have a need for styling attributes that change the entries depending on their context.

Not a huge issue, and if you feel that we should be permissive for more straightforward documentation, then I will resolve this one.

My worry is that if we accept more attributes that will require us to support them even if they might not be desired.

In the implementation will you support all of them?

Reply:
Yes, AFAIK all of them are supported, it's just that some are useless because they will not have any effect, due to the fact that a `toc-sequence` can currently only exist inside `pre-content` or `post-content`. When this limitation changes, the attributes could possibly become useful.

- `volume-keep-priority`
- `keep-with-previous-sheets`
- `keep-with-next-sheets`

In addition there is the `id` attribute which is of limited use inside a table of contents because it is either referenced by a `page-number` or a `toc-entry`. Referencing a `toc-block` from a `toc-entry`, either in the same or a different TOC, is adventurous to say the least. And also referencing a entry from a `page-number`, either inside or outside a TOC, is iffy because `toc-block` can be rendered multiple times so the reference could be ambiguous. The same can be said about any content inside `pre-content` or `post-content` by the way. I'm not even sure what will happen in Dotify if you try this.

Side note: I just noticed that `span` currently does not have an `id` attribute, but it is supported in the implementation (see https://github.com/brailleapps/dotify.formatter.impl/pull/44). I've created an issue: https://github.com/braillespecs/obfl/issues/85.

Reply:
Ok, @bertfrees.

Great explanation and discussion. I'm fine by resolving this and keeping the specification as-is. I was not sure that was the intent of your comment, so is there anything you believe we should change?

Reply:
Shall I remove the `block-atts` and add these:
`    <ref name="list-atts"/>
    <ref name="row-spacing"/>
    <ref name="block-atts-except-list-type-row-spacing-id"/>
`?
(This basically means that I omit the `block-id`.)

Reply:
I'm unsure. Since referring to an `id` within `pre-content` or `post-content` is never a good idea, not only within a `toc-sequence`, I'm actually inclined to say just leave it as is.

By the way I think we also have to make sure that the documentation and the schema matches because this is currently not the case and it has caused confusion.